lib/packwerk/const_node_inspector.rb

```diff
@@ -17,12 +17,10 @@ def constant_name_from_node(node, ancestors:)
       if constant_in_module_or_class_definition?(node, ancestors: ancestors)
         fully_qualify_constant(node, ancestors: ancestors)
       else
-        begin
-          Node.constant_name(node)
-        rescue Node::TypeError
-          nil
-        end
+        Node.constant_name(node)
       end
+    rescue Node::TypeError
```

Comment:
I’m on the fence about this one. The `rescue` was [originally added](https://github.com/Shopify/packwerk-old/commit/8315ad638d2ad414dd43569ac270542d92f4594d) to deal with an occurrence of `self.class::HEADERS` in core. I agree that `Node.constant_name` can raise `TypeError` elsewhere in this method but that’s not something we’ve seen happen so I’m not able to decide whether returning `nil` would be the right thing to do in those unknown cases.

Reply:
I'm going to remove this change from the PR because another PR was opened that deals with this issue more concretely https://github.com/Shopify/packwerk/pull/54